service/history/queues/executable.go

```diff
@@ -365,6 +384,20 @@ func (e *executableImpl) HandleErr(err error) (retErr error) {
 		return nil
 	}
 
+	var tsme TaskStateMismatchError
+	if errors.As(err, &tsme) {
+		metrics.TaskSkipped.With(e.taggedMetricsHandler).Record(1)
```

Comment:
Discussed in person.

Reply:
Let's wait until we refactor the other executors in a separate PR.
If we don't return `nil, nil` from `LoadMutableStateForTask` we can change the executor functions to avoid checking if mutable state is nil too.